src/main/java/com/shoppinglist/shoppinglist/dao/ShoppingListDAO.java

```diff
@@ -28,7 +28,7 @@ public List<ListItem> fetchAll() {
 
     @Override
     public ListItem fetch(int id) throws NoSuchElementException {
-        return shoppingListRepository.findById(id).orElseThrow();
+        return shoppingListRepository.findById(id).orElseThrow(() -> new NoSuchElementException("Item not found with ID: " + id));
     }

```

Comment:
Provides a customized error message that includes the ID of the item not found in the error message.